File: spring_boot_backend_template/src/main/java/com/project/Entity/Lesson.java
package com.project.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString(exclude = "course,admin")

public class Lesson extends BaseEntity{
	
	@Column(name="lesson_name",nullable = true)
	private String lessonName;
	
	@Column(name="source_link")
	 @NotBlank(message = "SourceLink cannot be empty")
	 private String sourceLink;
	
	@Column(name="video_link")
	@NotBlank(message = "VideoLink cannot be empty")
	 private String videoLink;

	    @ManyToOne
	    @JoinColumn(name = "course_id",nullable = false)
	    private Course course;

	    @ManyToOne
	    @JoinColumn(name = "user_id",nullable = false)
	    private User admin;

}

File: backend/Solution1/ELearn/Models/Lesson.cs
using System;
using System.Collections.Generic;

namespace ELearn.Models
{
    public partial class Lesson
    {
        public long Id { get; set; }
        public DateOnly? CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public string? LessonName { get; set; }
        public string SourceLink { get; set; } = null!;
        public string VideoLink { get; set; } = null!;
        public long AdminId { get; set; }
        public long CourseId { get; set; }

        public virtual User Admin { get; set; } = null!;
        public virtual Course Course { get; set; } = null!;
    }
}
